### PR TITLE
replace default generate_nonce with one that uses secure random

### PR DESCRIPTION
**Pull request by dshafer**

### CODE REVIEW

**Comment by smholloway** [comment]
:+1: :shipit:

File: lib/toopher-web/OAuth.php [hunk 1]
```diff
@@ -481,16 +481,18 @@ public function build_signature($signature_method, $consumer, $token) {
...
+   * util function: generate secure nonce
    */
-  private static function generate_nonce() {
-    $mt = microtime();
-    $rand = mt_rand();
-
-    return md5($mt . $rand); // md5s look nicer than numbers
+  private static function generate_nonce($size = 16) {
+    if (!function_exists('openssl_random_pseudo_bytes')) {
+      throw new \InvalidArgumentException('OpenSSL not supported.');
```
Comment (egrim) [inline lib/toopher-web/OAuth.php]: Can we please reword this to target the user's needs instead of the system's state (e.g. "Missing required dependency: OpenSSL")?

Comment (dshafer) [inline lib/toopher-web/OAuth.php]: Ok, but it's a bit more than a missing dependency - it means that their actual PHP was built without OpenSSL, so they need a whole new interpreter.  Maybe something like "This PHP was built without OpenSSL support.  Please install a PHP built with OpenSSL to use the Toopher-Wordpress plugin"

Comment (egrim) [inline lib/toopher-web/OAuth.php]: Sounds good - thanks!

And I assume `InvalidArgumentException` is idiomatic (despite there being no arguments involved)?

**Comment by egrim** [comment]
Some lingering concerns that we're throwing the wrong exception, but assuming that's what we do in PHP: :+1:

**Comment by dshafer** [comment]
There's also a builtin exception called [BadFunctionCallException](http://php.net/manual/en/class.badfunctioncallexception.php) that might be more appropriate.  I'll switch to that one.
